examples/generative/vq_vae.py

```diff
@@ -357,7 +366,23 @@ def show_subplot(original, reconstructed):
 print(f"Input shape of the PixelCNN: {pixelcnn_input_shape}")
 
 """
-Don't worry about the input shape. It'll become clear in the following sections.
+This input shape represents the MNIST image size reduction from 28x28 pixels in the original digit
+images, that pass through two layers of stride-2 convolution. With "same" padding, this exactly
+halves the "resolution" of the output shape for each stride-2 layer, so with these two layers we
+end up with an encoder output tensor of 7x7 on axes 2 and 3, with the first axis as the batch size
+and the last axis being the code book embedding size. Since the quantization layer in the
+autoencoder maps these 7x7 tensors to indices of the code book, these output layer axis sizes must
+be matched by the PixelCNN as the input shape. The task of the PixelCNN for this architecture is to
+generate _likely_ 7x7 arrangements of codebook indices.
+
+Note that this shape is something to optimize for in larger-sized image domains. Since the PixelCNN
+is autoregressive, it needs to pass over each codebook index sequentially in order to generate
+novel images from the codebook. Each stride-2 (or rather more correctly a stride (2, 2)) convolution
+layer will divide the image generation time by four. Note, however, that there is probably a lower
+bound on this part: when the number of codes for the image to reconstruct is too small, it has
+insufficient information for the decoder to represent the level of detail in the image, so the
+output quality will suffer. Finding the sweet spot for this trade-off can require some architecture
+tweaking and could very well differ per dataset.
```

Comment:
The first part of this could be amended to tell the readers that a larger codebook might be needed. That'd also make the point on trade-off stronger.

Reply:
I agree, I'll add a commit to mention this, thanks

Reply:
I updated this in https://github.com/keras-team/keras-io/pull/938/commits/6663681752049de96a75161c81a229ee1aaccd9f, I'd appreciate your thoughts on this.